### DISCUSSION TITLE
Debugging in IntelliJ

### DISCUSSION THREAD

**Discussion by dp200x** [nushell/nushell#8228] [Q&A]
Hi, thanks for creating this awesome project. I'm a rust newbie and would like to go through the execution of the shell using intellij debugger. Would like to know if there're instructions to run/debug through intellij.

**Comment by dp200x** [comment]
I've tried the following

<img width="1032" alt="Screenshot 2023-02-26 at 12 55 15 PM" src="https://user-images.githubusercontent.com/126346403/221427575-99251e5b-9d2c-4629-ad63-ebbd53e15f32.png">

but getting an error

```
ERROR: no TTY for interactive shell

Process finished with exit code 1
```

**Reply by sholderbach** [reply]
This apparently encounters the same problem as with the `code lldb` extension in VS Code #6788 . The terminal STDIN needs to appear as a useable terminal. Maybe try disabling the `Emulate terminal in output` console option or look for other options in regards to the hosting terminal.
